src/stdlib_math.fypp

```diff
@@ -1,21 +1,270 @@
 #:include "common.fypp"
 #:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
 
 module stdlib_math
     use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace
+    public :: EULERS_NUMBER_SP, EULERS_NUMBER_DP, EULERS_NUMBER_QP
+    public :: DEFAULT_LINSPACE_LENGTH, DEFAULT_LOGSPACE_BASE, DEFAULT_LOGSPACE_LENGTH
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    integer, parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace
+    real(sp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
 
     interface clip
     #:for k1, t1 in IR_KINDS_TYPES
         module procedure clip_${k1}$
     #:endfor
     end interface clip
 
+    interface linspace
+    !! Version: Experimental
+    !!
+    !! Create rank 1 array of linearly spaced elements
+    !! If the number of elements is not specified, create an array with size 100. If n is a negative value,
+    !! return an array with size 0. If n = 1, return an array whose only element is end
+    !!([Specification](../page/specs/stdlib_math.html#description))
+    #:for k1, t1 in RC_KINDS_TYPES
+      #:set RName = rname("linspace_default", 1, t1, k1)
+      module function ${RName}$(start, end) result(res)
+        ${t1}$, intent(in) :: start
+        ${t1}$, intent(in) :: end
+
+        ${t1}$ :: res(DEFAULT_LINSPACE_LENGTH)
+      end function ${RName}$
```

Comment:
@certik Should this definition of function include an `implicit none`?

Reply:
The implicit none at the top of the module should propagate into all functions declared within the module, should it not? I believe it does.

Reply:
I though like you until recently. Now I am not sure anymore. 
My question is related to [this comment](https://github.com/fortran-lang/benchmarks/pull/23#discussion_r660616297).

Reply:
Wow, that is horrifying. Where in the interface must you rewrite the implicit none? Inside each function declaration? Or just once at the beginning of the interface block?

Also, could you reapprove the workflow checks? It appears there was an error with MYSYS booting up, and I don't think it has anything to do with your revisions. I'd like to see if that's resolved before I make any more changes on my end.

Reply:
I found [this ](https://stackoverflow.com/a/24337734/11838135)SO discussion addressing this issue.